File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/GotoStatement.vb
Imports System

Namespace Tests.Diagnostics

    Public Class GotoStatement

        Sub GotoStatement(condition As Boolean)

            If condition Then
                GoTo Label 'Noncompliant {{Remove this use of 'GoTo'.}}
'               ^^^^
            Else
                Return
            End If

Label:
            Throw New Exception()

        End Sub

        Sub OnError()
            On Error GoTo nextstep ' Compliant, handled by S2359
nextstep:
            System.Console.WriteLine("Error")
        End Sub

    End Class

End Namespace

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/GotoStatement.cs
using System;
using System.Collections.Generic;

namespace Tests.Diagnostics
{
    public class GotoStatement
    {
        void foo(int a)
        {
            var @goto = 5;

            goto Label; //Noncompliant {{Remove this use of 'goto'.}}
//          ^^^^

            Label:
            ;

            int n = 5;
            switch (n)
            {
                case 1:
                    break;
                case 2:
                    goto default; //Noncompliant
//                  ^^^^
                case 3:
                    goto case 1; //Noncompliant
                default:
                    break;
            }
        }
    }
}
